File: src/main/java/xyz/emirdev/echogen/commands/SmiteCommand.java
package xyz.emirdev.echogen.commands;

import java.util.List;

import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import com.mojang.brigadier.context.CommandContext;
import com.mojang.brigadier.exceptions.CommandSyntaxException;
import com.mojang.brigadier.tree.LiteralCommandNode;

import io.papermc.paper.command.brigadier.CommandSourceStack;
import io.papermc.paper.command.brigadier.Commands;
import io.papermc.paper.command.brigadier.argument.ArgumentTypes;
import net.kyori.adventure.text.minimessage.tag.resolver.Placeholder;
import xyz.emirdev.echogen.PluginCommand;

public class SmiteCommand extends PluginCommand {

    @Override
    public LiteralCommandNode<CommandSourceStack> getCommand() {
        return Commands.literal("smite")
                .requires(hasPermission("echogen.smite"))
                .then(Commands.argument("players", ArgumentTypes.players())
                        .executes(this::smite))
                .build();
    }

    private int smite(CommandContext<CommandSourceStack> ctx) throws CommandSyntaxException {
        CommandSender sender = ctx.getSource().getSender();
        List<Player> targets = getPlayers(ctx, "players");
        if (targets == null)
            return 1;

        for (Player target : targets) {
            target.getWorld().strikeLightningEffect(target.getLocation());

            sender.sendRichMessage(
                    "<aqua>You have smitten <player>!</aqua>",
                    Placeholder.unparsed("player", target.getName()));
        }

        return 1;
    }
}

File: src/main/kotlin/cat/emir/echogen/commands/SmiteCommand.kt
package cat.emir.echogen.commands

import cat.emir.echogen.Echogen
import com.mojang.brigadier.context.CommandContext

import io.papermc.paper.command.brigadier.CommandSourceStack
import io.papermc.paper.command.brigadier.argument.ArgumentTypes
import cat.emir.echolib.command.PluginCommand
import cat.emir.echolib.command.getPlayers
import cat.emir.echolib.sendLangMessage
import com.mojang.brigadier.builder.LiteralArgumentBuilder

class SmiteCommand(plugin: Echogen) : PluginCommand<Echogen>(plugin) {
    override val aliases = setOf("thor")

    override fun getCommand(): LiteralArgumentBuilder<CommandSourceStack> {
        return command("smite") {
            requires { it.sender.hasPermission("echogen.smite") }
            argument("players", ArgumentTypes.players()) {
                executes(::smite)
            }
        }
    }

    fun smite(ctx: CommandContext<CommandSourceStack>): Int {
        val sender = ctx.source.sender
        val targets = ctx.getPlayers("players") ?: return 1

        for (target in targets) {
            target.world.strikeLightningEffect(target.location)
            sender.sendLangMessage("smite.executor", listOf("player" to target.name))
        }

        return 1
    }
}
